src/main/native/smithwaterman/IntelSmithWaterman.cc

```diff
@@ -87,13 +87,16 @@ JNIEXPORT jint JNICALL Java_com_intel_gkl_smithwaterman_IntelSmithWaterman_align
     env->ReleasePrimitiveArrayCritical(alt, alternate, 0);
     env->ReleasePrimitiveArrayCritical(cigar, cigarArray, 0);
 
-    if(offset == -1 || env->ExceptionCheck()) {
+    if(offset == -1) {
```

Comment:
I wanted to replace both of the arbitrary values used as error codes with constants but I haven't been able to track -1 down in the code. I don't know if this branch is dead or if this is a possible output of the algorithm. I decided not to change it since I don't know how this was intended to work.

Reply:
This was on old condition from the C code.